packages/esm-service-queues-app/src/queue-table/queue-entry-actions/queue-entry-actions.modal.tsx

```diff
@@ -287,89 +252,53 @@ export const QueueEntryActionModal: React.FC<QueueEntryActionModalProps> = ({
                   title={t('noPrioritiesConfigured', 'No priorities configured')}
                 />
               ) : (
-                <ContentSwitcher
-                  size="sm"
-                  selectedIndex={selectedPriorityIndex}
-                  onChange={(event) => {
-                    setSelectedPriorityUuid(event.name as string);
+                <RadioButtonGroup
+                  className={styles.radioButtonGroup}
+                  valueSelected={formState.selectedPriority}
+                  onChange={(uuid) => {
+                    setSelectedPriorityUuid(uuid);
                   }}>
                   {priorities?.map(({ uuid, display }) => (
-                    <Switch
-                      role="radio"
-                      name={uuid}
-                      text={
-                        uuid == queueEntry.priority.uuid
-                          ? t('currentValueFormatted', '{{value}} (Current)', {
-                              value: display,
-                              interpolation: { escapeValue: false },
-                            })
-                          : display
-                      }
+                    <RadioButton
                       key={uuid}
+                      name={display}
+                      labelText={
+                        <Tag
+                          className={`${priorities.findIndex((p) => p.uuid === uuid) === 1 && styles.orange} ${
```

Comment:
It looks like we're doing this array index check both here and in the `type` property below. Can we centralize this logic into a function? Also, consider using `classnames` for merging multiple classes conditionally.